arrow-core-data/src/main/kotlin/arrow/core/Ordering.kt

```diff
@@ -23,14 +27,107 @@ sealed class Ordering {
     GT -> GT
   }
 
+  fun hash(): Int = hashWithSalt(hashCode())
+
+  fun hashWithSalt(salt: Int): Int =
+    salt.hashWithSalt(hashCode())
+
+  fun compare(b: Ordering): Ordering = when (this) {
+    is LT -> when (b) {
+      is LT -> EQ
+      else -> GT
+    }
+    is GT -> when (b) {
+      is GT -> EQ
+      else -> LT
+    }
+    is EQ -> b
+  }
+
+  fun compareTo(b: Ordering): Int = compare(b).toInt()
+
+  fun eqv(other: Ordering): Boolean = compare(other) == EQ
+
+  fun lt(b: Ordering): Boolean = compare(b) == LT
+
+  fun lte(b: Ordering): Boolean = compare(b) != GT
+
+  fun gt(b: Ordering): Boolean = compare(b) == GT
+
+  fun gte(b: Ordering): Boolean = compare(b) != LT
+
+  fun max(b: Ordering): Ordering = if (gt(b)) this else b
+
+  fun min(b: Ordering): Ordering = if (lt(b)) this else b
+
+  fun sort(b: Ordering): Tuple2<Ordering, Ordering> =
+    if (gte(b)) Tuple2(this, b) else Tuple2(b, this)
+
+  fun empty(): Ordering = EQ
+
+  fun combine(b: Ordering): Ordering = this + b
+
+  fun show(): String = when (this) {
+    LT -> "LT"
+    GT -> "GT"
+    EQ -> "EQ"
+  }
+
   companion object {
     fun fromInt(i: Int): Ordering = when (i) {
       0 -> EQ
       else -> if (i < 0) LT else GT
     }
+
+    fun eq(): Eq<Ordering> = OrderingEq
+
+    fun hash(): Hash<Ordering> = OrderingHash
+
+    fun monoid(): Monoid<Ordering> = OrderingMonoid
+
+    fun order(): Order<Ordering> = OrderingOrder
+
+    fun semigroup(): Semigroup<Ordering> = OrderingMonoid
+
+    fun show(): Show<Ordering> = OrderingShow
   }
 }
 
 object LT : Ordering()
 object GT : Ordering()
 object EQ : Ordering()
+
+fun Collection<Ordering>.combineAll(): Ordering =
+  if (isEmpty()) OrderingMonoid.empty() else reduce { a, b -> a.combine(b) }
+
+@Suppress("EXTENSION_SHADOWED_BY_MEMBER")
+private object OrderingEq : Eq<Ordering> {
+  override fun Ordering.eqv(b: Ordering): Boolean =
```

Comment:
@nomisRev Do we still need to implement these typeclasses for ordering? Or is it enough with having the concrete methods in the Ordering data type?

Reply:
I think it's still useful to keep these typeclasses, since none of these typeclasses will be deprecated.